libres/lib/enkf/enkf_obs.cpp

```diff
@@ -1140,23 +1142,22 @@ stringlist_type *enkf_obs_alloc_matching_keylist(const enkf_obs_type *enkf_obs,
         return obs_keys;
 
     stringlist_type *matching_keys = stringlist_alloc_new();
-    char **input_keys;
-    int num_keys;
+    std::vector<std::string> input_keys;
     int obs_keys_count = stringlist_get_size(obs_keys);
 
-    util_split_string(input_string, " ", &num_keys, &input_keys);
-    for (int i = 0; i < num_keys; i++) {
-        const char *input_key = input_keys[i];
+    ert::split(input_string, ' ',
```

Comment:
Would it be better to have this return the list instead?

Reply:
What do you think @pinkwah ?

Reply:
As the function argument serves as the loop body, its actually a bit easier to drop the temporary vector here.
```cpp
ert::split(input_string, ' ',
               [&obs_keys](auto input_key) { 
        for (int j = 0; j < obs_keys_count; j++) {
            const char *obs_key = stringlist_iget(obs_keys, j);

            if (util_string_match(obs_key, input_key.c_str()) &&
                !stringlist_contains(matching_keys, obs_key))
                stringlist_append_copy(matching_keys, obs_key);
        }
});
```

Reply:
Thanks, fixed.